tools/tests/test_osqueryd.py

```diff
@@ -190,6 +190,35 @@ def test_9_hostid_instance(self):
         self.assertTrue(daemon.isAlive())
         daemon.kill()
 
+    @test_base.flaky
+    def test_10_daemon_with_watchdog(self):
+        # Test that killed workers are respawned by the watcher
+        if os.environ.get('SANITIZE') is not None:
+            return
+        daemon = self._run_daemon({
+            "allow_unsafe": True,
+            "disable_watchdog": False,
+            "ephemeral": True,
+            "disable_database": True,
+            "disable_logging": True,
+        })
+        self.assertTrue(daemon.isAlive())
+
+        # Check that the daemon spawned a child process
+        children = daemon.getChildren()
+        self.assertTrue(len(children) > 0)
+
+        # Kill only the child worker
+        os.kill(children[0], signal.SIGINT)
```

Comment:
Test 3 does all of the above, but sends the SIGINT to the daemon to make sure the child stops. Can we rename this: `test_3_daemon_lost_worker` or something similar, and place it next to the other test 3?